.github/workflows/js-tests.yml

```diff
@@ -0,0 +1,40 @@
+name: Javascript Tests
+
+on: [push]
+
+jobs:
+  js_tests:
+
+    runs-on: ubuntu-latest
+
+    permissions:
+      actions: read
+
+    defaults:
+      run:
+        shell: bash -ileo pipefail {0}
+
+    steps:
+      - name: Checkout Streamlit code
+        uses: actions/checkout@v3
+        with: 
+          persist-credentials: false
+          submodules: 'recursive'
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+      - uses: ./.github/actions/pre_cache
+      - uses: ./.github/actions/make_init
+      - name: Run make develop
+        run: |
+          source venv/bin/activate
+          make develop
+      - name: Run linters
+        run: |
+          source venv/bin/activate
+          # Run eslint as a standalone command to generate the test report.
+          PRE_COMMIT_NO_CONCURRENCY=true DISABLE=eslint pipenv run pre-commit run --show-diff-on-failure --color=always --all-files
+          make jslint
+      - name: Run frontend tests
+        run: make jstest
```

Comment:
We should consider whether we want to publish test results.
For example, we can use action [dorny/test-reporter](https://github.com/dorny/test-reporter) or upload tests result as [job summary](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#adding-a-job-summary).

Reply:
+1 to this, although I think it'd be fine for now to aim to initially get these actions to behavioral parity with what currently runs in CircleCI, and we can consider whether to publish test results more nicely in followup changes